hack/testing-olm-upgrade/upgrade-common

```diff
@@ -397,15 +397,18 @@ discover_versions(){
   major_version="$(echo $full_version | cut -d'.' -f1)"
   minor_version="$(echo $full_version | cut -d'.' -f2)"
 
-  # we shouldn't run into this yet... maybe for 6.0 but we should like
-  # add the 'Replaces' spec field and use that for the previous_version
-  if [[ $minor_version -eq 0 ]]; then
-    log::info "Will be unable to calculate the previous_version since our minor version is 0"
-    exit 0
+  export version="$(echo $major_version.$minor_version)"
+
+  # Hard rewrite of 5.0.0 to superficial 4.7.0, as replacement for
+  # the breaking point when OpenShift Logging shifted release schema.
+  if [[ $major_version -eq 5 ]] && [[ $minor_version -eq 0 ]]; then
+    # Set this to 4.7 because we need to calculate the latest previous
+    # version prior to 5.0, which is 4.6 for OpenShift Logging.
+    major_version=4
+    minor_version=7
   fi
 
-  export version="$(echo $major_version.$minor_version)"
-  get_latest_previous_version $version
+  get_latest_previous_version "$(echo $major_version.$minor_version)"
```

Comment:
you can use the variable declared on line 400

Reply:
This is not possible because the `version` in L400 should is exported as `5.0` and should stay as such. However the need here is for a hard overwrite of 5.0 to 4.7 to make L411 work.

Reply:
ok